src/MobxPromise.ts

```diff
@@ -142,6 +149,13 @@ export class MobxPromiseImpl<R> {
     }
 
     @computed get result(): R | undefined {
+
+        // we need to
+        if (this.isComplete === false) {
+            /*@ts-ignore*/
+            if (this.onPrematureRead) this.onPrematureRead.call(this);
```

Comment:
@onursumer this is a little nicer.   i need to be able to execute some stuff when this happens so i can now configure that in code by tacking on a method to prototype